tools/run_tests/xds_k8s_test_driver/framework/xds_k8s_testcase.py

```diff
@@ -494,6 +495,26 @@ def cleanup(self):
         self.server_runner.cleanup(force=self.force_cleanup,
                                    force_namespace=self.force_cleanup)
 
+    def _check_pod_restart_time(self) -> None:
+        # We should fail test if pod restarted during test (b/269192257).
+        logger.info('Checking Pods restart times')
+        client_restarts = k8s.get_pod_restarts(self.client_runner.k8s_namespace,
+                                               self.client_runner.deployment)
+        self.assertEqual(
```

Comment:
We should not assert here, just get the data - so the cleanup can happen. After the cleanup, we should do the assertions, in a `finally` blocks.